### PR TITLE
Improve tinyfiledialogs integration

### PR DESCRIPTION
**Pull request by Meakk**
- Add file filters based on available plugins
- Fix default path on Windows

### CODE REVIEW

File: application/F3DStarter.cxx [hunk 1]
```diff
@@ -1624,7 +1624,24 @@ void F3DStarter::AddCommands()
       const char* file = std::getenv("CTEST_OPEN_DIALOG_FILE");
       if (!file)
       {
-        file = tinyfd_openFileDialog("Open File", ".", 0, nullptr, nullptr, false);
+        std::vector<std::string> filters;
+        for (const auto info : f3d::engine::getReadersInfo())
+        {
+          for (const auto ext : info.Extensions)
```
Comment (snoyer) [inline application/F3DStarter.cxx:1630]:
```suggestion
        for (const auto& info : f3d::engine::getReadersInfo())
        {
          for (const auto& ext : info.Extensions)
```
unless there's a good not to?

Comment (Meakk) [reply application/F3DStarter.cxx:1630]: right!

File: application/F3DStarter.cxx [hunk 2]
```diff
@@ -1624,7 +1624,24 @@ void F3DStarter::AddCommands()
       const char* file = std::getenv("CTEST_OPEN_DIALOG_FILE");
       if (!file)
```
Comment (mwestphal) [inline application/F3DStarter.cxx:1625]: you could consider adding the logic above the if to get coverage for this line, even if its not really tested, at least the code would be executed.

**Comment by Meakk** [comment]
FYI @hant-hub

**Review by mwestphal — APPROVED** [review]

**Comment by Meakk** [comment]
@mwestphal I added a commit, please check again

**Pull request merged**